lib/spack/spack/build_systems/cuda.py

```diff
@@ -104,7 +104,10 @@ def cuda_flags(arch_list):
     # This implies that the last one in the list has to be updated at
     # each release of a new cuda minor version.
     conflicts('%gcc@10:', when='+cuda ^cuda@:11.0')
-    conflicts('%gcc@11:', when='+cuda ^cuda@:11.4')
+    conflicts('%gcc@11:', when='+cuda ^cuda@:11.4.0')
+    conflicts('%clang@12:', when='+cuda ^cuda@:11.4.2')
+    conflicts('%gcc@12:', when='+cuda ^cuda@:11.5.0')
```

Comment:
```suggestion
    conflicts('%gcc@12:', when='+cuda ^cuda@:11.5.0')
    conflicts('%clang@12:', when='+cuda ^cuda@:11.4.2')
```

group by compiler. @ax3l i haven't verified whether this is correct or not.